lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -9,18 +9,16 @@
 	/**
 	 * Whether a theme or its parent have a theme.json file.
 	 *
-	 * @param boolean $clear_cache Whether the cache should be cleared and theme support recomputed. Default is false.
+	 * The result would be cached via the WP_Object_Cache
+	 * under the `wp_theme_has_theme_json` key.
 	 *
 	 * @return boolean
 	 */
-	function wp_theme_has_theme_json( $clear_cache = false ) {
-		static $theme_has_support = null;
-
-		if ( true === $clear_cache ) {
-			$theme_has_support = null;
-		}
-
-		if ( null !== $theme_has_support ) {
+	function wp_theme_has_theme_json() {
+		$cache_key         = 'wp_theme_has_theme_json';
+		$cache_found       = false;
+		$theme_has_support = wp_cache_get( $cache_key, '', false, $cache_found );
```

Comment:
Shouldn't we introduce a new cache group here? I haven't seen core caches outside of a cache group, maybe they exist, but at least I haven't come across one myself. I would be in favor of either using an existing cache group this fits into (e.g. `themes`) or creating a new one (maybe `block-themes` or something like that).